File: LastHMS2/Models/Bill.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace LastHMS2.Models
{
    public class Bill
    {
        [Key]
        public int Bill_Id { get; set; }
        [Display(Name = "Examination")]
        [Column(TypeName = "money")]
        public decimal? Bill_Examination { get; set; } // Currency

        [Column(TypeName = "money")]
        [Display(Name = "Surgeries")]
        public decimal? Bill_Surgeries { get; set; }

        [Column(TypeName = "money")]
        [Display(Name = "Rays")]
        public decimal? Bill_Rays { get; set; }

        [Column(TypeName = "money")]
        [Display(Name = "Medical Test")]
        public decimal? Bill_Medical_Test { get; set; }

        [Column(TypeName = "money")]
        [Display(Name = "Room Service")]
        public decimal? Bill_Room_Service { get; set; }

        [Column(TypeName = "money")]
        [Display(Name = "Medication")]
        public decimal? Bill_Medication { get; set; }
        
        [Display(Name = "Bill Date")]
        public DateTime Bill_Date{ get; set; }
        [Display(Name = "Is Paid")]
        public bool Paid { get; set; } // bayed
        [Required]
        public int Patient_Id { get; set; } // Foreign Key //cascade
    }
}

File: vikings/lib/models/bill.dart
class Bill {
  int? billId;
  late String billDate;
  double? billExamination;
  double? billSurgeries;
  double? billRays;
  double? billMedicalTest;
  double? billMedicalRoomService;
  double? billMedicalMedication;
  double? billTotal;
  late bool paid;

  Bill(
      {this.billId,
      required this.billDate,
      this.billExamination,
      this.billMedicalMedication,
      this.billMedicalRoomService,
      this.billMedicalTest,
      this.billRays,
      this.billSurgeries,
      this.billTotal,
      required this.paid});

  Bill.fromJson(Map bill) {
    billId = bill['Bill_Id'];
    billDate = bill['Bill_Date'];
    billExamination = bill['Bill_Examination'];
    billMedicalMedication = bill['Bill_Medication'];
    billMedicalRoomService = bill['Bill_Room_Service'];
    billMedicalTest = bill['Bill_Medical_Test'];
    billRays = bill['Bill_Rays'];
    billSurgeries = bill['Bill_Surgeries'];
    billTotal = bill['Bill_Id'] +
        bill['Bill_Date'] +
        bill['Bill_Examination'] +
        bill['Bill_Medication'] +
        bill['Bill_Room_Service'] +
        bill['Bill_Medical_Test'] +
        bill['Bill_Rays'] +
        bill['Bill_Surgeries'];
    paid=bill['Paid'];
  }
}
